openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/changeEvent/AbstractEventConsumer.java

```diff
@@ -0,0 +1,316 @@
+/*
+ *  Copyright 2021 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.openmetadata.service.apps.bundles.changeEvent;
+
+import static org.openmetadata.schema.entity.events.SubscriptionStatus.Status.ACTIVE;
+import static org.openmetadata.schema.entity.events.SubscriptionStatus.Status.AWAITING_RETRY;
+import static org.openmetadata.schema.entity.events.SubscriptionStatus.Status.FAILED;
+import static org.openmetadata.service.events.subscription.AlertUtil.getFilteredEvent;
+import static org.openmetadata.service.events.subscription.AlertUtil.getInitialAlertOffsetFromDb;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.entity.events.AlertMetrics;
+import org.openmetadata.schema.entity.events.EventSubscription;
+import org.openmetadata.schema.entity.events.EventSubscriptionOffset;
+import org.openmetadata.schema.entity.events.FailedEvent;
+import org.openmetadata.schema.entity.events.SubscriptionStatus;
+import org.openmetadata.schema.type.ChangeEvent;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.events.errors.EventPublisherException;
+import org.openmetadata.service.events.subscription.AlertUtil;
+import org.openmetadata.service.util.JsonUtils;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.Job;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.PersistJobDataAfterExecution;
+
+@Slf4j
+@DisallowConcurrentExecution
+@PersistJobDataAfterExecution
+public abstract class AbstractEventConsumer implements Consumer<ChangeEvent>, Job {
+  public static final String ALERT_OFFSET_KEY = "alertOffsetKey";
+  public static final String ALERT_INFO_KEY = "alertInfoKey";
+  public static final String OFFSET_EXTENSION = "eventSubscription.Offset";
+  public static final String METRICS_EXTENSION = "eventSubscription.metrics";
+  public static final String FAILED_EVENT_EXTENSION = "eventSubscription.failedEvent";
+  private int offset = -1;
```

Comment:
offset should be Long and we should store that way in db

Reply:
done